taui/src/components/form.js

```diff
@@ -99,26 +99,33 @@ class Form extends React.PureComponent<Props> {
       return
     }
     const position = destination.location.position
+    // check for older profiles that contain destination purposes not saved with
+    // the 'TripPurpose.' prefix required by react-i18next
+    const translatablePurpose = destination.purpose.startsWith('TripPurpose')
+      ? destination.purpose
+      : 'TripPurpose.' + destination.purpose
     return (position.lat !== 0 && position.lon !== 0) ? {
-      label: destination.purpose + ': ' + destination.location.label,
+      label: destination.location.label,
       position: position,
-      value: position
+      value: translatablePurpose
     } : null
   }
 
   setStateNetwork = (networks, userProfile) => {
     const useNetworks = this.getProfileNetworks(networks, userProfile)
     const first = useNetworks[0]
-    const network = {label: first.name, value: first.url}
+    const network = {label: first.name, value: first.url, networkName: first.name}
     this.setState({network})
-    this.props.setActiveNetwork(network.label)
+    this.props.setActiveNetwork(network.name)
   }
 
   selectDestination = (option?: ReactSelectOption) => {
     const destinationObj = option ? {
-      label: option.label,
+      // restore the label to contain only the address and remove translated
+      // destination name for re-generation
+      label: option.label.split(':')[1],
```

Comment:
I'm not sure whether this is necessary -- I couldn't find a place where `destination.label` is getting used. Does it break anything to revert this change and use `option.label` here?